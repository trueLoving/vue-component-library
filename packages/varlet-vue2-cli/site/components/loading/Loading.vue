<template>
  <div class="var-site-loading">
    <div class="var-site-loading__content" :class="[loading ? 'var-site-loading__content--active' : null]" v-if="hasSlots()">
      <slot />
    </div>
    <div class="var--box var-site-loading__body" :class="[hasSlots() ? 'var-site-loading__inside' : null]" v-if="isShow">
      <div class="var-site-loading__circle" v-if="type === 'circle'">
        <span
          class="var-site-loading__circle-block"
          :style="{
            width: getRadius * 2 + 'px',
            height: getRadius * 2 + 'px',
            color,
          }"
        >
          <svg viewBox="25 25 50 50">
            <circle cx="50" cy="50" r="20" fill="none"></circle>
          </svg>
        </span>
      </div>

      <template v-for="(nums, key) in loadingTypeDict">
        <div :key="key" :class="`var-site-loading__${key} var-site-loading__${key}-${size}`" v-if="type === key">
          <div
            v-for="num in nums"
            :key="num + key"
            :style="{ backgroundColor: color }"
            :class="`var-site-loading__${key}-item var-site-loading__${key}-item-${size}`"
          ></div>
        </div>
      </template>
      <div
        class="var-site-loading__description"
        :class="`var-site-loading__description--${size}`"
        :style="{ color }"
        v-if="hasSlots('description') || description"
      >
        <slot name="description">{{ description }}</slot>
      </div>
    </div>
  </div>
</template>

<script>
import { props } from './props'
import { toNumber } from '../utils/shared'
import { defineComponent } from '../utils/create'

export default defineComponent({
  name: 'VarSiteLoading',

  props,

  data: () => ({
    loadingTypeDict: {
      wave: 5,
      cube: 4,
      rect: 8,
      disappear: 3,
    },
    sizeDict: {
      mini: 9,
      small: 12,
      normal: 15,
      large: 18,
    },
  }),

  computed: {
    getRadius() {
      return this.radius ? toNumber(this.radius) : this.sizeDict[this.size]
    },

    isShow() {
      if (!this.hasSlots()) return true

      return this.loading
    },
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './loading';
</style>
