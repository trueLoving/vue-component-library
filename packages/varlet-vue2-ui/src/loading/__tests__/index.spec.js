import example from '../example'
import Loading from '..'
import VarLoading from '../Loading'
import { mount } from '@vue/test-utils'
import Vue from 'vue'

test('test loading example', () => {
  const wrapper = mount(example)

  expect(wrapper.html()).toMatchSnapshot()
})

test('test loading plugin', () => {
  Vue.use(Loading)
  expect(Vue.component(Loading.name)).toBeTruthy()
})

test('test circle props', async () => {
  const wrapper = mount(VarLoading, {
    propsData: {
      type: 'circle',
      color: '#2979ff',
      radius: '12',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test wave props', async () => {
  const wrapper = mount(VarLoading, {
    propsData: {
      type: 'wave',
      color: '#2979ff',
      size: 'small',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test cube props', async () => {
  const wrapper = mount(VarLoading, {
    propsData: {
      type: 'cube',
      color: '#2979ff',
      size: 'small',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test rect props', async () => {
  const wrapper = mount(VarLoading, {
    propsData: {
      type: 'rect',
      color: '#2979ff',
      size: 'small',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test disappear props', async () => {
  const wrapper = mount(VarLoading, {
    propsData: {
      type: 'disappear',
      color: '#2979ff',
      size: 'small',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test slots', async () => {
  const template = `
    <var-loading description="loading...." type="circle" :loading="loading">
      <div>test</div>
    </var-loading>
  `
  const wrapper = mount({
    components: {
      [VarLoading.name]: VarLoading,
    },
    data() {
      return {
        loading: false,
      }
    },
    template,
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setData({ loading: true })

  expect(wrapper.html()).toMatchSnapshot()
})
