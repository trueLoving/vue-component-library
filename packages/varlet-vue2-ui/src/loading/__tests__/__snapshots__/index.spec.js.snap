// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test loading example 1`] = `
"<div class=\\"var-loading-example\\">
  <div class=\\"loading-demo__type\\">
    <div class=\\"app-type\\">类型</div>
    <div class=\\"loading-demo__type-block\\">
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <div class=\\"var-loading__cube var-loading__cube-normal\\">
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <div class=\\"var-loading__wave var-loading__wave-normal\\">
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"var-loading__rect var-loading__rect-normal\\">
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <!---->
          <!---->
          <div class=\\"var-loading__disappear var-loading__disappear-normal\\">
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div class=\\"loading-demo__type\\">
    <div class=\\"app-type\\">颜色</div>
    <div class=\\"loading-demo__type-block\\">
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px; color: rgb(41, 121, 255);\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <div class=\\"var-loading__cube var-loading__cube-normal\\">
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
          </div>
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <div class=\\"var-loading__wave var-loading__wave-normal\\">
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
          </div>
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"var-loading__rect var-loading__rect-normal\\">
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <!---->
          <!---->
          <div class=\\"var-loading__disappear var-loading__disappear-normal\\">
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div class=\\"loading-demo__size\\">
    <div class=\\"app-type\\">大小</div>
    <div class=\\"loading-demo__size-block\\">
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 24px; height: 24px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <div class=\\"var-loading__cube var-loading__cube-small\\">
            <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <div class=\\"var-loading__wave var-loading__wave-small\\">
            <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"var-loading__rect var-loading__rect-small\\">
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <div class=\\"var-loading\\">
        <!---->
        <div class=\\"var--box var-loading__body\\">
          <!---->
          <!---->
          <!---->
          <!---->
          <div class=\\"var-loading__disappear var-loading__disappear-small\\">
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-small\\" style=\\"background-color: currentColor;\\"></div>
            <div class=\\"var-loading__disappear-item var-loading__disappear-item-small\\" style=\\"background-color: currentColor;\\"></div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class=\\"app-type\\">包裹内容</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\" style=\\"margin-bottom: 8px;\\">
      <!---->
      <div class=\\"var-button__content\\">
        打开
      </div>
    </button>
    <div class=\\"var-loading\\">
      <div class=\\"var-loading__content\\">
        <div class=\\"var-card var-elevation--2\\">
          <!---->
          <div class=\\"var-card__title\\">介绍</div>
          <!---->
          <div class=\\"var-card__description\\">Varlet-vue2 是一个基于Vue2开发的 Material 风格移动端组件库，全面拥抱Vue2生态，由社区的小伙伴开发和维护。</div>
          <!---->
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`test circle props 1`] = `
"<div class=\\"var-loading\\">
  <!---->
  <div class=\\"var--box var-loading__body\\">
    <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 24px; height: 24px; color: rgb(41, 121, 255);\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`test wave props 1`] = `
"<div class=\\"var-loading\\">
  <!---->
  <div class=\\"var--box var-loading__body\\">
    <!---->
    <div class=\\"var-loading__wave var-loading__wave-small\\">
      <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__wave-item var-loading__wave-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
    </div>
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`test cube props 1`] = `
"<div class=\\"var-loading\\">
  <!---->
  <div class=\\"var--box var-loading__body\\">
    <!---->
    <!---->
    <div class=\\"var-loading__cube var-loading__cube-small\\">
      <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__cube-item var-loading__cube-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
    </div>
    <!---->
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`test rect props 1`] = `
"<div class=\\"var-loading\\">
  <!---->
  <div class=\\"var--box var-loading__body\\">
    <!---->
    <!---->
    <!---->
    <div class=\\"var-loading__rect var-loading__rect-small\\">
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__rect-item var-loading__rect-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
    </div>
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`test disappear props 1`] = `
"<div class=\\"var-loading\\">
  <!---->
  <div class=\\"var--box var-loading__body\\">
    <!---->
    <!---->
    <!---->
    <!---->
    <div class=\\"var-loading__disappear var-loading__disappear-small\\">
      <div class=\\"var-loading__disappear-item var-loading__disappear-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__disappear-item var-loading__disappear-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
      <div class=\\"var-loading__disappear-item var-loading__disappear-item-small\\" style=\\"background-color: rgb(41, 121, 255);\\"></div>
    </div>
    <!---->
  </div>
</div>"
`;

exports[`test slots 1`] = `
"<div class=\\"var-loading\\">
  <div class=\\"var-loading__content\\">
    <div>test</div>
  </div>
  <!---->
</div>"
`;

exports[`test slots 2`] = `
"<div class=\\"var-loading\\">
  <div class=\\"var-loading__content var-loading__content--active\\">
    <div>test</div>
  </div>
  <div class=\\"var--box var-loading__body var-loading__inside\\">
    <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <!---->
    <!---->
    <!---->
    <!---->
    <div class=\\"var-loading__description var-loading__description--normal\\" style=\\"color: currentColor;\\">loading....</div>
  </div>
</div>"
`;